phpstan.neon

```diff
@@ -1,2 +1,12 @@
 parameters:
-  level: 1
+  level: 6
+  phpVersion: 70430 # PHP 7.4.30
+  ignoreErrors:
+  -
+    message: "#^If condition is always true.$#"
+    count: 1
+    path: lib/Loop/Loop.php
+  -
+    message: "#^Left side of && is always true.$#"
+    count: 1
+    path: lib/Loop/Loop.php
```

Comment:
which lines were affected?

Reply:
```
------ ------------------------------ 
  Line   lib/Loop/Loop.php             
 ------ ------------------------------ 
  72     If condition is always true.  
  178    Left side of && is always true.  
 ------ --------------------------------- 
```

Reply:
seems legit, thx for the hint

Reply:
```
    public function setInterval(callable $cb, float $timeout): array
    {
        $keepGoing = true;
        $f = null;

        $f = function () use ($cb, &$f, $timeout, &$keepGoing) {
            if ($keepGoing) {
                $cb();
                $this->setTimeout($f, $timeout);
            }
        };
        $this->setTimeout($f, $timeout);

        // Really the only thing that matters is returning the $keepGoing
        // boolean value.
        //
        // We need to pack it in an array to allow returning by reference.
        // Because I'm worried people will be confused by using a boolean as a
        // sort of identifier, I added an extra string.
        return ['I\'m an implementation detail', &$keepGoing];
    }
```
I guess that `$cb` can modify the value at the address `&$keepGoing` - but `phpstan` has no way to really know that?

Maybe I should add a phpstan "ignore line" tag with a comment?

Reply:
phpstan has some open problems regarding references. I think its not worth the effort

Reply:
```
    public function run(): void
    {
        $this->running = true;

        do {
            $hasEvents = $this->tick(true);
        } while ($this->running && $hasEvents);
        $this->running = false;
    }
```

`$this->running` is "always true" inside the loop. It gets set to `false` in the `stop()` method. I guess that there is some way for `stop()` to happen? And phpstan has no way understand that?

Reply:
> phpstan has some open problems regarding references. I think its not worth the effort

yeh, I will just leave it in the ignored-errors list. A later phpstan release might come along and "magically" not report this any more.